chainerio/cache/file_cache.py

```diff
@@ -77,21 +77,33 @@ def unlock(self):
 
 
 class FileCache(cache.Cache):
-    '''Stores cache data in local filesystem
+    '''Cache system with local filesystem
 
-    Stores cache data in local filesystem,
+    Stores cached data in local filesystem,
     ``~/.chainer/chainerio/cache`` by default. Cache data is
-    automatically deleted after the object is collected. If the
-    process quit (typically by SIGTERM) the cache remains after the
-    death of process.
+    automatically deleted after the object is collected. When this
+    object is not correctly closed, (e.g., the process killed by
+    SIGTERM), the cache remains after the death of process.
+
+    Arguments:
+        length (int): Length of the cache array.
+        multithread_safe (bool): Defines multithread safety. If this
+            is ``True``, reader-writer locking is introduced behind
```

Comment:
A readers-writer lock is introduced

Reply:
https://github.com/chainer/chainerio/pull/63#discussion_r327914793

Reply:
Mentioning lock primitive and design is not only for overhead. 1) When designing a multi-threaded system, the locking primitive behind it is very important. For example in Python, `asyncio.Lock` and `threading.Lock` can't be mixed without *practical* level of carefulness. Also it is not easily safe to mix with `multiprocessing.Lock`. Thus I added  description on what primitive is used.
2) The type of lock is important for app developer, too. Understanding that `get` doesn't block another `get` but `put` blocks `get` is important to design multithreaded and read-intensive program like ours. The design would change a lot if it is just a single mutex, where `get` blocks `get`.

Reply:
Mentioning lock primitive and design is not only for overhead. 1) When designing a multi-threaded system, the locking primitive behind it is very important. For example in Python, `asyncio.Lock` and `threading.Lock` can't be mixed without *practical* level of carefulness. Also it is not easily safe to mix with `multiprocessing.Lock`. Thus I added  description on what primitive is used.
2) The type of lock is important for app developer, too. Understanding that `get` doesn't block another `get` but `put` blocks `get` is important to design multithreaded and read-intensive program like ours. The design would change a lot if it is just a single mutex, where `get` blocks `get`.